source/agora/consensus/protocol/Nominator.d

```diff
@@ -1197,8 +1197,10 @@ extern(D):
                 // for all the candidates might take long.
                 version (unittest)
                 if (auto msg = this.ledger.validateConsensusData(data))
-                    assert(0, format!"combineCandidates: Invalid consensus data: %s"(
-                        msg));
+                {
+                    log.error("combineCandidates: Invalid consensus data: {}", msg);
+                    continue;
```

Comment:
In fact it probably should not skip adding as this does not happen in production version.